### PR TITLE
FIX: Remove measures with attributes name 'id' or 'type' (hotfix 3.4.3)

### PR DESCRIPTION
**Pull request by AlvaroVega**
Issue https://github.com/telefonicaid/iotagent-node-lib/issues/1485

this should be tested 
- [x] with https://github.com/telefonicaid/iotagent-node-lib/pull/1482/
- [x]  curl -i -X POST 'http://localhost:7897/iot/json?i=disp1&k=idmaootoin375kl9iifz7472z' -d '{"m": 3, "id": "mierda", "type":"otro"}' -H 'content-type: application/json'

### CODE REVIEW

**Comment by mapedraza** [comment]
> We should include at same point of the documentation (any suggestion @mapedraza ?) something like this:
> 
> > Note that `id` and `type` are not valid attribute names at Context Broker. Thus, although a `id` or `type` will not break the IOT Agent, they are silently ignored and never progress toward Context Broker entities.
> 
> or something with a similar meaning.

Probably, instead of deleting, renaming them to `_id` or `_type` could be useful in the future.

**Comment by fgalan** [comment]
> Probably, instead of deleting, renaming them to `_id` or `_type` could be useful in the future

It depends on how strict we want to be with "illegal" measures. Deleting them will strongly disallow its use, while moving them to attributes `_id` and `_type` will somehow allow its usage.

In my opinion it's better to delete, but it's not a strong opinion.

**Comment by mapedraza** [comment]
LGTM

**Review by mapedraza — APPROVED** [review]

File: lib/services/ngsi/entities-NGSI-v2.js [hunk 1]
```diff
@@ -340,6 +340,15 @@ function sendUpdateValueNgsi2(entityName, attributes, typeInformation, token, ca
         attributes,
         typeInformation
     );
+    // if any attribute name of a measure is 'id' or 'type' should be removed
```
Comment (fgalan) [inline lib/services/ngsi/entities-NGSI-v2.js:343]:
```suggestion
    // if any measure has name 'id' or 'type' it should be removed
    // (as they cannot be progressed as attribute names, given that 'id' or 'type' are forbidden names for attributes in CB)
```

File: CHANGES_NEXT_RELEASE [hunk 2]
```diff
@@ -1 +1 @@
-
+- FIX: remove attribute of measures with name `id` or `type` to avoid collisions
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]:
```suggestion
```

If this PR combines bump (note packages.json has been also modified) the CNR should be flushed. However, I'll keep this sentence for the future release notes in github.

File: doc/api.md [hunk 3]
```diff
@@ -179,7 +179,9 @@ information coming from the device is mapped to the Context Broker attributes:
 All of them have the same syntax, a list of objects with the following attributes:
 
 -   **object_id** (optional): name of the attribute as coming from the device.
--   **name** (mandatory): ID of the attribute in the target entity in the Context Broker.
+-   **name** (mandatory): ID of the attribute in the target entity in the Context Broker. Note that `id`` and `type``
```
Comment (fgalan) [inline doc/api.md:182]:
```suggestion
-   **name** (mandatory): ID of the attribute in the target entity in the Context Broker. Note that `id` and `type`
```

File: doc/api.md [hunk 4]
```diff
@@ -179,7 +179,9 @@ information coming from the device is mapped to the Context Broker attributes:
 All of them have the same syntax, a list of objects with the following attributes:
 
 -   **object_id** (optional): name of the attribute as coming from the device.
--   **name** (mandatory): ID of the attribute in the target entity in the Context Broker.
+-   **name** (mandatory): ID of the attribute in the target entity in the Context Broker. Note that `id` and `type`
+    are not valid attribute names at Context Broker. Thus, although a id or type will not break the IOT Agent, they
```
Comment (fgalan) [inline doc/api.md:183]:
```suggestion
    are not valid attribute names at Context Broker. Thus, although a measure named `id` or `type` will not break the IOT Agent, they 
```

File: doc/api.md [hunk 5]
```diff
@@ -209,6 +211,10 @@ Additionally for commands (which are attributes of type `command`) the following
 -   **contentType**: `content-type` header used when send command by HTTP transport (ignored in other kinds of
     transports)
 
+Note that, when information comming from devices, this means measures, are not defined neither in the group, nor in the 
+device, the IoT agent will store that information into the destination entity using the same attribute name than the 
+measure name, unless `explicitAttrs` is defined. Measures `id` or `type` names are invaled, and will be ignored.
```
Comment (fgalan) [inline doc/api.md:216]:
```suggestion
measure name, unless `explicitAttrs` is defined. Measures `id` or `type` names are invalid, and will be ignored. 
```

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**